### DISCUSSION TITLE
Any example for RPC DiscoveryService?

### DISCUSSION THREAD

**Discussion by Evgen1000end** [activej/activej#207] [Q&A] [answered]
I found interface DiscoveryService but there is no documentation with simple example how use it.

i have following code in my client:

```
void pollingAddresses() {
 // for example, it may be eureka, consul etc
  InetAddress[] newAddresses = InetAddress.getAllByName(host);
}
```

How correct notify DiscoveryService/RPCStrategy about update?

**Comment by eduard-vasinskyi** [comment, marked as answer]
Hi, @Evgen1000end

`DiscoveryService` in RPC is kinda an experimental feature. It will be greatly reworked/simplified in ActiveJ v6.0.
As for now, there is only one implementation of a dynamic discovery service which is a [`SettableDiscoveryService`](https://github.com/activej/activej/blob/master/cloud-rpc/src/main/java/io/activej/rpc/client/sender/SettableDiscoveryService.java). 
To use this implementation of a discovery service you need to first create a new `SettableDiscoveryService` with initial RPC server addresses.
Then you would need to create an [`RpcStrategyList`](https://github.com/activej/activej/blob/master/cloud-rpc/src/main/java/io/activej/rpc/client/sender/RpcStrategyList.java) out of this discovery service:
```java
RpcStrategyList.ofDiscoveryService(discoveryService);
```
Finally, you need to create an actual `RpcStrategy` that uses this `RpcStrategyList` and set this strategy to the Rpc client:
```java
RpcClient.create(eventloop)
	. . .
     .withStrategy(RpcStrategies.roundRobin(rpcStrategyList))
```
Whenever addresses of RPC servers change, you would need to call `SettableDiscoveryService#setAddressMap` from within the Eventloop thread to set new addresses. RPC client will then connect to new addresses and will abandon irrelevant connections.
